### PR TITLE
fix: set reply's default charset to utf8

### PR DESCRIPTION
**Pull request by xtx1130**
ref: https://github.com/fastify/fastify/issues/3788

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by xtx1130** [comment]
benchmark:

this branch:
image
master:
image

File: test/internals/reply.test.js [hunk 1]
```diff
@@ -729,6 +729,30 @@ test('non-string with content type application/json SHOULD be serialized as json
...
+
+  const fastify = require('../..')()
+
+  fastify.get('/', function (req, reply) {
+    reply.type('application/json; version=2; ').send({ key: 'hello world!' })
+  })
+
+  fastify.listen({ port: 0 }, err => {
+    t.error(err)
+    fastify.server.unref()
```
Comment (Eomm) [inline test/internals/reply.test.js:743]: I know it is used in other tests, but could you use the `t.teaddown` function instead of `server.unref`?

Comment (xtx1130) [reply test/internals/reply.test.js]: PTAL, but I want to know what is the difference between `t.teardown` & `server.unref`, more semantic?

**Review by mcollina — APPROVED** [review]
Good spot! This is an amazing speed bump. How did you find it?

**Review by zekth — APPROVED** [review]
Nice :v:

File: test/internals/reply.test.js (hunk 1 above)
Comment (climba03003) [reply test/internals/reply.test.js]: `t.teardown` run when the test end.
`server.unref` just didn't block the program end when no code need to run inside the event loop. It is an unclean close.

**Comment by xtx1130** [comment]
> Good spot! This is an amazing speed bump. How did you find it?

Thanks for the bug report, and I find these codes can be more simplified

File: docs/Reference/Reply.md [hunk 2]
```diff
@@ -261,6 +261,7 @@ Sets the content type for the response. This is a shortcut for
 ```js
 reply.type('text/html')
 ```
+When set `Content-Type` with json type, if you don't set `charset`, it will use `utf-8` by default.
```
Comment (Fdawgs) [inline docs/Reference/Reply.md:264]: Have to go into a meeting now but just stating that this doesn't read well.

Will review and update this comment when i'm out!

**Review by delvedor — APPROVED** [review]
🚀

File: docs/Reference/Reply.md (hunk 2 above)
Comment (mcollina) [reply docs/Reference/Reply.md:264]: Ooops, I missed this. I'll amend on main.

**Pull request merged**